### PR TITLE
[AIRFLOW-5842] Switch to Debian buster image. Depends on [AIRFLOW-5704]

### PR DESCRIPTION
**Pull request by potiuk**
NOTE FOR REVIEWER: This change depends on  #6373 . Please check only last commit.

As part of the production migration we switch to buster image
which is latest released LTS version.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5842

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [X] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -263,84 +255,70 @@ ENV AIRFLOW_REPO=${AIRFLOW_REPO}
...
+# Increase the value here to force reinstalling Apache Airflow pip dependencies
+ARG PIP_DEPENDENCIES_EPOCH_NUMBER="1"
+ENV PIP_DEPENDENCIES_EPOCH_NUMBER=${PIP_DEPENDENCIES_EPOCH_NUMBER}
 
-# Note! We are copying everything with airflow:airflow user:group even if we use root to run the scripts
-# This is fine as root user will be able to use those dirs anyway.
+# In case of CI builds we want to pre-install master version of airflow dependencies so that
+# We do not have to always reinstall it from the scratch and loose time for that.
+# CI build is optimised for build speed
+RUN pip install --user \
```
Comment (mik-laj) [inline Dockerfile:285]: Should we add `--no-cache-dir` here?

Comment (mik-laj) [reply Dockerfile:285]: Oh. I have now seen that you set environment variables on the 55 line, but maybe you should delete the variables and make it more implicit? It also remove 1 docker layer.

**Review by mik-laj — APPROVED** [review]

**Comment by potiuk** [comment]
Please don't

File: BREEZE.rst [hunk 2]
```diff
@@ -79,11 +79,11 @@ images maintained on the Docker Hub in the ``apache/airflow`` repository.
...
 There are three images that we are currently managing:
 
-* **CI image*** that is used for testing od both Unit tests and static check tests.
-  It contains a lot test-related packages (size of ~1GB).
-  Its tag follows the pattern of ``<BRANCH>-python<PYTHON_VERSION>-ci``
+* **CI image*** that is used for testing. It contains a lot more test-related installed software
+  (size of ~1GB). Its tag follows the pattern of ``<BRANCH>-python<PYTHON_VERSION>-ci``
   (for example, ``apache/airflow:master-python3.6-ci``). The image is built using the
   `<Dockerfile>`_ Dockerfile.
+* **Production-ready image** that is used TODO
```
Comment (feluelle) [inline BREEZE.rst]: Can we have more information on this?

**Review by feluelle — APPROVED** [review]
@potiuk What is the status on this?

**Comment by potiuk** [comment]
Getting there shortly! I added the automated test to be able to fix some of the difficulties with bash scripts testing and I am full steam ahead with improving that.

**Comment by potiuk** [comment]
Closing as I restarted it from the scratch (and simpler) - #7647

**Pull request closed without merging**